lms/djangoapps/course_api/blocks/forms.py

```diff
@@ -0,0 +1,129 @@
+        except ValueError:
+            raise ValidationError("'{}' is not a valid depth value.".format(value))
+
+    def clean_usage_key(self):
+        """
+        TODO
+        """
+        usage_key = self.cleaned_data['usage_key']
+
+        try:
+            usage_key = UsageKey.from_string(usage_key)
+            usage_key = usage_key.replace(course_key=modulestore().fill_in_run(usage_key.course_key))
+        except InvalidKeyError:
+            raise ValidationError("'{}' is not a valid usage key.".format(unicode(usage_key)))
+
+        return usage_key
+
+    def clean_return_type(self):
+        """
+        Return valid 'return_type' or default value of 'dict'
+        """
+        return self.cleaned_data['return_type'] or 'dict'
+
+    def clean_requested_user(self, cleaned_data, course_key):
+        """
+        Validates and returns the requested_user, while checking permissions.
+        """
+        requested_username = cleaned_data.get('user', '')
+        requesting_user = self.initial['requesting_user']
+
+        if requesting_user.username.lower() == requested_username.lower():
+            requested_user = requesting_user
+        else:
+            # the requesting user is trying to access another user's view
+            # verify requesting user can access another user's blocks
+            if not can_access_other_users_blocks(requesting_user, course_key):
+                raise PermissionDenied(
+                    "'{requesting_username}' does not have permission to access view for '{requested_username}'."
+                    .format(requesting_username=requesting_user.username, requested_username=requested_username)
+                )
+
+            # update requested user object
+            try:
+                requested_user = User.objects.get(username=requested_username)
+            except User.DoesNotExist:
+                raise Http404("Requested user '{username}' does not exist.".format(username=requested_username))
+
+        # verify whether the requested user's blocks can be accessed
+        if not can_access_users_blocks(requested_user, course_key):
+            raise PermissionDenied(
+                "Course blocks for '{requested_username}' cannot be accessed."
+                .format(requested_username=requested_username)
+            )
+
+        return requested_user
+
+    def clean(self):
+        cleaned_data = super(BlockListGetForm, self).clean()
+
+        # add additional requested_fields that are specified as separate parameters, if they were requested
+        additional_requested_fields = [
+            'student_view_data',
+            'block_counts',
+            'nav_depth',
+        ]
+        for additional_field in additional_requested_fields:
+            field_value = cleaned_data.get(additional_field)
+            if field_value or field_value == 0:  # allow 0 as a requested value
+                cleaned_data['requested_fields'].add(additional_field)
```

Comment:
By the way, there's a `test_nav_depth_invalid` unit test that makes sure the the integer validation happens.